Crash when fetching hidden files list

## Exception
* __User Action:__ UI Error
* __Request:__ Application crash
* __Version:__ 3.5.2
* __OS:__ Linux Android 10 - 29
* __Device:__ raphael
* __Model:__ Mi 9T Pro
* __Product:__ raphael_eea
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
at android.os.AsyncTask$4.done(AsyncTask.java:399)
at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
at java.util.concurrent.FutureTask.run(FutureTask.java:271)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.IllegalStateException: Unexpected failure to classify SearchResult: SearchResult{key=/storage/emulated/0/Pictures/Infinity/interestingasfuck-mb7.jpg, nodeFound=Node{edge=/storage/emulated/0/Pictures/Infinity/halo-kbewb0.mp4, value=-, edges=[]}, charsMatched=38, charsMatchedInNodeFound=38, parentNode=Node{edge=, value=null, edges=[Node{edge=/storage/emulated/0/Pictures/Infinity/halo-kbewb0.mp4, value=-, edges=[]}]}, parentNodesParent=null, classification=null}
at com.googlecode.concurrenttrees.radix.ConcurrentRadixTree$SearchResult.classify(ConcurrentRadixTree.java:989)
at com.googlecode.concurrenttrees.radix.ConcurrentRadixTree$SearchResult.<init>(ConcurrentRadixTree.java:969)
at com.googlecode.concurrenttrees.radix.ConcurrentRadixTree.searchTree(ConcurrentRadixTree.java:932)
at com.googlecode.concurrenttrees.radix.ConcurrentRadixTree.getValueForExactKey(ConcurrentRadixTree.java:102)
at com.amaze.filemanager.utils.DataUtils.isFileHidden(DataUtils.java:359)
at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.createListParcelables(LoadFilesListTask.java:273)
at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.lambda$doInBackground$4(LoadFilesListTask.java:231)
at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.lambda$doInBackground$4$LoadFilesListTask(Unknown Source:0)
at com.amaze.filemanager.asynchronous.asynctasks.-$$Lambda$LoadFilesListTask$UXzmWslcH0cfZfAA6Y9hqkVwFiA.invoke(Unknown Source:6)
at com.amaze.filemanager.filesystem.root.ListFilesCommand.getFilesList(ListFilesCommand.kt:173)
at com.amaze.filemanager.filesystem.root.ListFilesCommand.listFiles(ListFilesCommand.kt:67)
at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.doInBackground(LoadFilesListTask.java:222)
at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.doInBackground(LoadFilesListTask.java:67)
at android.os.AsyncTask$3.call(AsyncTask.java:378)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
... 3 more

```
</details>
<hr>

Another mail:

```
{"request":"Application crash","user_action":"UI Error","product":"WW_X00TD","package":"com.amaze.filemanager","os":"Linux Android 10 - 29","model":"ASUS_X00TD","time":"2020-12-05 00:43","version":"3.5.1","device":"ASUS_X00T_3","exceptions":"[java.lang.RuntimeException: Unable to resume activity {com.amaze.filemanager\/com.amaze.filemanager.ui.activities.MainActivity}: java.lang.IllegalStateException: Unexpected failure to classify SearchResult: SearchResult{key=\/storage\/emulated\/0, nodeFound=Node{edge=emulated\/0, value=0, edges=[]}, charsMatched=19, charsMatchedInNodeFound=10, parentNode=Node{edge=\/storage\/, value=null, edges=[Node{edge=CEB2-27BF, value=1, edges=[]}, Node{edge=emulated\/0, value=0, edges=[]}]}, parentNodesParent=Node{edge=, value=null, edges=[Node{edge=\/storage\/, value=null, edges=[Node{edge=CEB2-27BF, value=1, edges=[]}, Node{edge=emulated\/0, value=0, edges=[]}]}]}, classification=null}\n\tat android.app.ActivityThread.performResumeActivity(ActivityThread.java:4206)\n\tat android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4238)\n\tat android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)\n\tat android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)\n\tat android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)\n\tat android.app.ActivityThread$H.handleMessage(ActivityThread.java:2017)\n\tat android.os.Handler.dispatchMessage(Handler.java:107)\n\tat android.os.Looper.loop(Looper.java:214)\n\tat android.app.ActivityThread.main(ActivityThread.java:7397)\n\tat java.lang.reflect.Method.invoke(Native Method)\n\tat com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)\n\tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:935)\nCaused by: java.lang.IllegalStateException: Unexpected failure to classify SearchResult: SearchResult{key=\/storage\/emulated\/0, nodeFound=Node{edge=emulated\/0, value=0, edges=[]}, charsMatched=19, charsMatchedInNodeFound=10, parentNode=Node{edge=\/storage\/, value=null, edges=[Node{edge=CEB2-27BF, value=1, edges=[]}, Node{edge=emulated\/0, value=0, edges=[]}]}, parentNodesParent=Node{edge=, value=null, edges=[Node{edge=\/storage\/, value=null, edges=[Node{edge=CEB2-27BF, value=1, edges=[]}, Node{edge=emulated\/0, value=0, edges=[]}]}]}, classification=null}\n\tat com.googlecode.concurrenttrees.radix.ConcurrentRadixTree$SearchResult.classify(ConcurrentRadixTree.java:989)\n\tat com.googlecode.concurrenttrees.radix.ConcurrentRadixTree$SearchResult.<init>(ConcurrentRadixTree.java:969)\n\tat com.googlecode.concurrenttrees.radix.ConcurrentRadixTree.searchTree(ConcurrentRadixTree.java:932)\n\tat com.googlecode.concurrenttrees.radix.ConcurrentRadixTree.putInternal(ConcurrentRadixTree.java:456)\n\tat com.googlecode.concurrenttrees.radix.ConcurrentRadixTree.put(ConcurrentRadixTree.java:83)\n\tat com.googlecode.concurrenttrees.radixinverted.ConcurrentInvertedRadixTree.put(ConcurrentInvertedRadixTree.java:185)\n\tat com.amaze.filemanager.utils.DataUtils.putDrawerMetadata(DataUtils.java:416)\n\tat com.amaze.filemanager.ui.views.drawer.Drawer.addNewItem(Drawer.java:608)\n\tat com.amaze.filemanager.ui.views.drawer.Drawer.refreshDrawer(Drawer.java:299)\n\tat com.amaze.filemanager.ui.activities.MainActivity.onResume(MainActivity.java:1184)\n\tat android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1454)\n\tat android.app.Activity.performResume(Activity.java:7939)\n\tat android.app.ActivityThread.performResumeActivity(ActivityThread.java:4196)\n\t... 11 more\n]","user_comment":""} 
```

--- a/app/src/main/java/com/amaze/filemanager/utils/DataUtils.java
+++ b/app/src/main/java/com/amaze/filemanager/utils/DataUtils.java
@@ -40,6 +40,7 @@ import com.googlecode.concurrenttrees.radixinverted.ConcurrentInvertedRadixTree;
 import com.googlecode.concurrenttrees.radixinverted.InvertedRadixTree;
 
 import android.text.TextUtils;
+import android.util.Log;
 import android.view.MenuItem;
 
 import androidx.annotation.Nullable;
@@ -356,7 +357,12 @@ public class DataUtils {
   }
 
   public boolean isFileHidden(String path) {
-    return getHiddenFiles().getValueForExactKey(path) != null;
+    try {
+      return getHiddenFiles().getValueForExactKey(path) != null;
+    } catch (IllegalStateException e) {
+      Log.w(getClass().getSimpleName(), e);
+      return false;
+    }
   }
 
   public ConcurrentRadixTree<VoidValue> getHiddenFiles() {

